### PR TITLE
SUL23-120 | @jdwjdwjdw | Structure and Style Places to Study Card

### PR DESCRIPTION
**Pull request by jdwjdwjdw**
# READY FOR REVIEW

# Summary
- [SUL23-120](https://stanfordits.atlassian.net/browse/SUL23-120): FE-DEV | Style Places to Study card
Screen Shot 2023-03-31 at 5 24 46 PM
Screen Shot 2023-03-31 at 5 25 12 PM

# Review By (Date)
- When convenient

# Urgency
- How critical is this PR?

# Steps to Test

1. Checkout branch
2. Create a couple of branch locations
3. Edit the Places of Study Taxonomy so that terms are added
4. Create Places of Study based on those branch locations
5. Create basic page with Places of Study List
6. Review to make sure it matches designs
7. Review code

Three current known issues/questions:
- I updated the lists paragraph to be full width for Places of Study, but I am struggling at fixing all the other lists to not be full width (see https://deploy-preview-66--stanford-library.netlify.app/sws-test-329-example-lists)
- The Modal for places to study only shows the image every other time you click it, which appears to be related to the resize hook, as the Modal only seems to recognize `width` half the time. If needed, removing that hook altogether still results in an acceptable experience using breakpoints only.
- I'm not sure what is needed in the page-display file. For now I just have it saying `Place of Study`

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- [SUL23-120](https://stanfordits.atlassian.net/browse/SUL23-120): FE-DEV | Style Places to Study card

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

[SUL23-120]: https://stanfordits.atlassian.net/browse/SUL23-120?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/(public)/components/node/sul-study-place/card.tsx [hunk 1]
```diff
@@ -1,10 +1,73 @@
...
+                  {features && features.slice(0,4).map((feature, index) =>
+                    <li key={`feature-${index}`} className="su-type-1 su-leading-display">
+                      {feature.name}
+                    </li>
+                  )}
+                </ul>
+              }
+
+              {(features && features.length > 4) &&
+                <StudyPlaceModal node={node} />
```
Comment (pookmish) [inline app/(public)/components/node/sul-study-place/card.tsx:63]: Try to get in the practice of only passing the necessary parts down, especially when you're in a client side component. You probably only actually need 5 or 6 props from the node, not the entire thing.

File: app/(public)/components/node/sul-study-place/study-place-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      <Modal
+        isOpen={modalOpen}
+        onClose={() => setModalOpen(false)}
+        ariaLabel="Features Modal"
+      >
+        <div className={"su-bg-white su-flex su-w-full su-leading-display su-shadow-md su-border-0 su-rounded su-flex-row"}>
+          <div className="su-hidden md:su-block su-rs-px-3 su-rs-py-3 su-w-1/2">
+            <div className={"su-overflow-hidden su-aspect-[4/3] su-relative "}>
+              {contactImage}
+            </div>
```
Comment (pookmish) [inline app/(public)/components/node/sul-study-place/study-place-modal.tsx:51]: Modify this to match [this example](https://github.com/SU-SWS/sulgryphon-nextjs/pull/66/files#diff-279548defe2331652ddec986a58e0bfae2d071669886070a11569789dbfd17eaR18)

File: app/(public)/components/node/sul-study-place/study-place-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import Image from "next/image";
+import Modal from "../../patterns/modal";
+import {ErrorBoundary} from "react-error-boundary";
+import {ChevronRightIcon} from "@heroicons/react/20/solid";
+import {MapPinIcon } from "@heroicons/react/24/outline";
+import LibCal from "./libcal";
+import StudyPlaceHours from "./study-place-today-hours";
+import Conditional from "@/components/utils/conditional";
+
+const StudyPlaceModal = ({node}: { node: StudyPlace }) => {
```
Comment (pookmish) [inline app/(public)/components/node/sul-study-place/study-place-modal.tsx:15]: as mentioned, take in only the props you actually need.

File: app/(public)/components/node/sul-study-place/study-place-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            <div className={"su-overflow-hidden su-aspect-[4/3] su-relative "}>
+              {contactImage}
+            </div>
+
+            <Conditional showWhen={node.sul_study__libcal_id}>
+              <LibCal libcalId={node.sul_study__libcal_id}/>
+            </Conditional>
+          </div>
+
+          <div className={"card-body su-items-start su-rs-px-3 su-rs-pb-3 su-rs-pt-7 md:su-rs-pt-3 su-w-full "}>
```
Comment (pookmish) [inline app/(public)/components/node/sul-study-place/study-place-modal.tsx:58]: Don't need the brackets here or the extra space at the end of the class list.

File: app/(public)/components/node/sul-study-place/study-place-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            <div className="su-leading-display su-text-18 su-pt-0 su-font-normal ">
+              <h2 className="su-type-3 su-rs-mb-1">{node.sul_study__type.name}</h2>
```
Comment (pookmish) [inline app/(public)/components/node/sul-study-place/study-place-modal.tsx:60]: Should these be `h2` or `h3`... I'm not exactly sure...

File: app/(public)/components/node/sul-study-place/study-place-modal.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                    <div>{node.title}</div>
+                  </Link>	
+                </div>
+
+                {(node.sul_study__capacity || features) &&
+                  <ul className="su-ml-10 su-rs-mb-1">
+                    {node.sul_study__capacity &&
+                      <li className="su-type-1 su-leading-display">{node.sul_study__capacity.name}</li>
+                    }
+                    {features && features.slice(0,4).map((feature, index) =>
```
Comment (pookmish) [inline app/(public)/components/node/sul-study-place/study-place-modal.tsx:79]: You don't want to slice them when we're in the modal. It should display all features.

File: app/(public)/components/node/sul-study-place/card.tsx [hunk 7]
```diff
@@ -1,10 +1,73 @@
...
+
+              {(node.sul_study__capacity || features) &&
+                <ul className="su-ml-10 su-rs-mb-1">
+                  {node.sul_study__capacity &&
+                    <li className="su-type-1 su-leading-display">{node.sul_study__capacity.name}</li>
+                  }
+                  {features && features.slice(0,4).map((feature, index) =>
+                    <li key={`feature-${index}`} className="su-type-1 su-leading-display">
+                      {feature.name}
+                    </li>
```
Comment (pookmish) [inline app/(public)/components/node/sul-study-place/card.tsx:57]:
```suggestion
                  {features && features.slice(0, 4).map(feature =>
                    <li key={`feature-${feature.id}`} className="su-type-1 su-leading-display">
                      {feature.name}
                    </li>
```

**Review by pookmish — CHANGES_REQUESTED** [review]

File: app/(public)/components/node/sul-study-place/study-place-modal.tsx (hunk 5 above)
Comment (pookmish) [reply app/(public)/components/node/sul-study-place/study-place-modal.tsx:60]: Yeah, i'm not really sure. Just a thought to think about. We can review it in during an a11y review

**Comment by pookmish** [comment]
FYI, I rebased your branch to fix the merge conflicts. to sync your work do `git fetch; git reset --hard origin feature/SUL23-120--places-to-study-card`

File: app/(public)/components/node/sul-study-place/study-place-modal.tsx (hunk 6 above)
Comment (jdwjdwjdw) [reply app/(public)/components/node/sul-study-place/study-place-modal.tsx]: Ahh yes, good catch.

**Review by pookmish — APPROVED** [review]

**Pull request merged**